### PR TITLE
Add a plugin util method to find and insert hosts to meta

### PR DESCRIPTION
**Pull request by bzwei**
This helper method is included in ansible-rulebook which can be used in any source plugin codebase.

### CODE REVIEW

File: ansible_rulebook/plugin_utils.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        hosts = dpath.get(event_data, host_path, path_separator)
+    except KeyError:
+        # does not contain host
+        return
+
+    if isinstance(hosts, str):
+        if host_separator:
+            hosts = ",".join(hosts.split(host_separator))
+    elif isinstance(hosts, list) or isinstance(hosts, tuple):
+        hosts = ",".join(hosts)
```
[the remark this answers is not in the document]
Comment (bzwei) [reply ansible_rulebook/plugin_utils.py]: I now converted hosts to list type for all scenarios.

File: ansible_rulebook/plugin_utils.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Any, Dict, NoReturn
+
+import dpath
+
+
+def insert_hosts_to_meta(
+    event_data: Dict[str, Any], args: Dict[str, Any]
```
[the remark this answers is not in the document]
Comment (bzwei) [reply ansible_rulebook/plugin_utils.py:21]: This function reads data_host_path, data_path_separator, data_host_separator from the plugin initialize args. They are provided by users.

File: ansible_rulebook/plugin_utils.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+#  Copyright 2022 Red Hat, Inc.
```
Comment (mkanoor) [inline ansible_rulebook/plugin_utils.py:1]: @bzwei Should this be 2023?

File: ansible_rulebook/plugin_utils.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+import dpath
+
+
+def insert_hosts_to_meta(
+    event_data: Dict[str, Any], args: Dict[str, Any]
+) -> NoReturn:
+    """Find hosts and insert them to the meta dict in an event
+
+    When ansible-rulebook runs a playbook it automatically fetches hosts in
```
Comment (mkanoor) [inline ansible_rulebook/plugin_utils.py:25]: @bzwei 
When-ansible-rulebook runs a rulebook?

**Comment by mkanoor** [comment]
@bzwei Shouldn't this be used as a source-plugin and be present in the eda.ansible collection?
https://github.com/ansible/event-driven-ansible/tree/main/plugins/event_filter

**Comment by bzwei** [comment]
Replaced by https://github.com/ansible/event-driven-ansible/pull/84.

**Pull request closed without merging**